bee-api/bee-rest-api/src/endpoints/routes/api/v2/mod.rs

```diff
@@ -22,141 +22,34 @@ pub mod treasury;
 pub mod utxo_changes_by_milestone_id;
 pub mod utxo_changes_by_milestone_index;
 
-use std::net::IpAddr;
+use axum::Router;
 
-use bee_gossip::NetworkCommandSender;
-use bee_ledger::workers::consensus::ConsensusWorkerCommand;
-use bee_protocol::workers::{config::ProtocolConfig, MessageSubmitterWorkerEvent, PeerManager};
-use bee_runtime::{node::NodeInfo, resource::ResourceHandle};
-use bee_tangle::Tangle;
-use tokio::sync::mpsc;
-use warp::{self, Filter, Rejection, Reply};
+use crate::endpoints::storage::StorageBackend;
 
-use crate::endpoints::{config::RestApiConfig, storage::StorageBackend, Bech32Hrp, NetworkId};
+pub(crate) const MAX_RESPONSE_RESULTS: usize = 1000;
 
-pub(crate) fn path() -> impl Filter<Extract = (), Error = warp::Rejection> + Clone {
-    super::path().and(warp::path("v2"))
-}
-
-#[allow(clippy::too_many_arguments)]
-pub(crate) fn filter<B: StorageBackend>(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    tangle: ResourceHandle<Tangle<B>>,
-    storage: ResourceHandle<B>,
-    message_submitter: mpsc::UnboundedSender<MessageSubmitterWorkerEvent>,
-    network_id: NetworkId,
-    bech32_hrp: Bech32Hrp,
-    rest_api_config: RestApiConfig,
-    protocol_config: ProtocolConfig,
-    peer_manager: ResourceHandle<PeerManager>,
-    network_command_sender: ResourceHandle<NetworkCommandSender>,
-    node_info: ResourceHandle<NodeInfo>,
-    consensus_worker: mpsc::UnboundedSender<ConsensusWorkerCommand>,
-) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
-    add_peer::filter(
-        public_routes.clone(),
-        allowed_ips.clone(),
-        peer_manager.clone(),
-        network_command_sender.clone(),
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().nest(
+        "/v2",
+        add_peer::filter::<B>()
+            .merge(info::filter::<B>())
+            .merge(message::filter::<B>())
```

Comment:
Can you put the module to plural like in the URL please? It's a bit inconsistent atm, you have peers but have message and output...

Reply:
fixed